velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -2281,3 +2281,97 @@ TEST_F(StringFunctionsTest, concatInSwitchExpr) {
       {"This is a long sentence-zzz"_sv, "aaa-This is some other sentence"_sv});
   test::assertEqualVectors(expected, result);
 }
+
+TEST_F(StringFunctionsTest, fromBigEndian32) {
+  const auto fromBigEndian32 = [&](std::optional<std::string> arg) {
+    return evaluateOnce<int32_t, std::string>(
+        "from_big_endian_32(c0)", {arg}, {VARBINARY()});
+  };
+
+  EXPECT_EQ(std::nullopt, fromBigEndian32(std::nullopt));
+  EXPECT_THROW(fromBigEndian32(hexToDec("01")), VeloxUserError);
+  EXPECT_THROW(fromBigEndian32(hexToDec("0000000000000001")), VeloxUserError);
+
+  EXPECT_EQ(0, fromBigEndian32(hexToDec("00000000")));
+  EXPECT_EQ(1, fromBigEndian32(hexToDec("00000001")));
+  EXPECT_EQ(-1, fromBigEndian32(hexToDec("FFFFFFFF")));
+  EXPECT_EQ(12345678, fromBigEndian32(hexToDec("00BC614E")));
+  EXPECT_EQ(-12345678, fromBigEndian32(hexToDec("FF439EB2")));
+  EXPECT_EQ(2147483647, fromBigEndian32(hexToDec("7FFFFFFF")));
+  EXPECT_EQ(-2147483647, fromBigEndian32(hexToDec("80000001")));
+  EXPECT_EQ(-2147483648, fromBigEndian32(hexToDec("80000000")));
+}
+
+TEST_F(StringFunctionsTest, toBigEndian32) {
+  const auto toBigEndian32 = [&](std::optional<int32_t> arg) {
+    return evaluateOnce<std::string, int32_t>(
+        "to_big_endian_32(c0)", {arg}, {INTEGER()});
+  };
+
+  EXPECT_EQ(std::nullopt, toBigEndian32(std::nullopt));
+
+  EXPECT_EQ(hexToDec("00000000"), toBigEndian32(0));
+  EXPECT_EQ(hexToDec("00000001"), toBigEndian32(1));
+  EXPECT_EQ(hexToDec("FFFFFFFF"), toBigEndian32(-1));
+  EXPECT_EQ(hexToDec("00BC614E"), toBigEndian32(12345678));
+  EXPECT_EQ(hexToDec("FF439EB2"), toBigEndian32(-12345678));
+  EXPECT_EQ(hexToDec("7FFFFFFF"), toBigEndian32(2147483647));
+  EXPECT_EQ(hexToDec("80000001"), toBigEndian32(-2147483647));
+  EXPECT_EQ(hexToDec("80000000"), toBigEndian32(-2147483648));
+}
+
+TEST_F(StringFunctionsTest, fromBigEndian64) {
```

Comment:
Please add test cases for boundary values (2^64, 2^63  and -(2^63), -(2^64))

Reply:
These examples are already included below, and 2^64 and -(2^64) are outside of the int64_t range. Is there anything else that needs to be added?

Reply:
It will be good to see the overflow being triggered in the test. Lets add those in any case.

Reply:
Yes, I added overflow tests for the functions.